core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -323,7 +346,18 @@ public void clearCellContent(EObject object) {
         if (element.getTarget() instanceof DAnnotation) {
           DAnnotation annotation = (DAnnotation) element.getTarget();
           if (annotation.getSource() != null && annotation.getSource().equals(type)) {
-            list.add(annotation);
+            if(type.equals(TitleBlockHelper.ELEMENT_TITLE_BLOCK)) {
+              DNodeContainer node = (DNodeContainer) element;
+              if(!node.getIncomingEdges().isEmpty()) {
+                DEdge edge = node.getIncomingEdges().iterator().next();

```

Comment:
could use .get(0) instead  of iterator:-??

Reply:
Done.